implementations/code/code_changes_git.py

```diff
@@ -0,0 +1,78 @@
+from datetime import datetime
+from itertools import groupby
+import conditions
+import utils
+from commit_git import CommitGit
+
+
+class CodeChangesGit(CommitGit):
+    """
+    Class for Code_Changes for Git repositories.
+    """
+
+    def compute(self):
+        """
+        Count number of commits of different types, like including
+        empty commits or counting only those commits made on
+        the master branch.
+
+        :returns count: the number of commits satisfying the conditions passed
+            while instantiating CodeChangesGit.
+        """
+
+        commit_hashes = [item['hash'] for item in self.items]
+        return len(set(commit_hashes))
+
+    def _agg(self, items, period):
+        """
+        Perform an aggregation operation on a DataFrame or Series
+        to count the number of commits created in a every interval
+        of the period specified in the time_series method, like
+        'M', 'W',etc.
+
+        It simply counts the number of rows in the series, excluding
+        NaN values.
+
+        :param items: A list of dictionaries on which to apply the aggregation.
+
+        :param period: A string which can be any one of the pandas time
+        series rules:
+            'W': week
+            'M': month
+            'D': day
+
+        :returns count: the number of commits in the given series.
+        """
+
+        timeseries = dict()
+        items_groupby = groupby(
+            items, key=lambda x: (x['created_date'].year,
+                                  x['created_date'].month))
+
+        for period, group in items_groupby:
+            group = list(group)
+
+            timeseries[period] = len(group)
+
```

Comment:
I'm having a little trouble generating a timeseries similar to the result of the `resample()` method in the `pandas` implementation.
- For example, `df.resample('M').count()` performs the count aggregation operation on the groups but also returns a value of 0 for those groups with no elements. It is hard to do this with the `itertools` I am using.
- `df.resample()` allows for easily passing required timeperiods like `M` or `W`. Though I can use `itertool` to groupby month like the current implementation I have done, I'll have to come up with a different way to do it, so that it is easier for a period of a `week` or `day`.

Reply:
First of all, we don't need to do all the fancy stuff that Pandas resample does. We can stick to the basics, or even forget about the timeseries stuff, since we really don't need it for showing the definition of the metric.

I suggest that we start without timeseries, and then in a separate pull request, we discuss about the best way of integrating it. If we find a reasonable way, that's it. If not, we can live without that functionality for this "branch" of the reference implementation.

In any case, for having months with zero items, you can just iterate after you have the dictionary with all months with non-zero items, filling in the gaps. You can do that at the same time you convert the data to a list (which maybe is the best representation for the resulting time serie). That is, you iterate from first month to last month, and if there is nothing in the dictonary for that month, the result is zero. If there is, the result is that one.

You can also try to do that when you build the dictionary, using itertools to iterate on the list of months, but that's a bit complex and I m not sure it's doable without coding it myself.

Reply:
> I suggest that we start without timeseries, and then in a separate pull request, we discuss about the best way of integrating it. If we find a reasonable way, that's it. If not, we can live without that functionality for this "branch" of the reference implementation.

Sure. I am removing the functionality for now.
Thanks for the other suggestions as well!